src/net/emitter.hpp

```diff
@@ -15,24 +15,42 @@ class Emitter : public Transport {
   public:
     void emit_connect() override {
         logger->log(MK_LOG_DEBUG2, "emitter: emit 'connect' event");
+        if (!do_connect) {
+            logger->log(MK_LOG_DEBUG2, "emitter: no handler set; ignoring");
+            return;
+        }
```

Comment:
This set of changes is again to increase robustness. This is to make sure that, if we have set callbacks to `nullptr` thus saying *we are not interested*, we do not receive unwanted notifications causing exceptions.